components/domains/renewal.tsx

```diff
@@ -0,0 +1,285 @@
+  }, [email, usState, salt]);
+
+  useEffect(() => {
+    if (!selectedDomains) return;
+    setPrice(
+      getPriceFromDomains(
+        selectedDomainsToArray(selectedDomains),
+        duration
+      ).toString()
+    );
+  }, [selectedDomains]);
+
+  useEffect(() => {
+    if (userBalanceDataError || !userBalanceData) setBalance("");
+    else {
+      const high = userBalanceData?.["balance"].high << BigInt(128);
+      setBalance((userBalanceData?.["balance"].low + high).toString(10));
+    }
+  }, [userBalanceData, userBalanceDataError]);
+
+  useEffect(() => {
+    if (balance && price) {
+      if (gweiToEth(balance) > gweiToEth(price)) {
+        setInvalidBalance(false);
+      } else {
+        setInvalidBalance(true);
+      }
+    }
+  }, [balance, price]);
+
+  function changeEmail(value: string): void {
+    setEmail(value);
+    setEmailError(isValidEmail(value) ? false : true);
+  }
+
+  useEffect(() => {
+    if (isUsResident) {
+      salesTax.getSalesTax("US", usState).then((tax) => {
+        setSalesTaxRate(tax.rate);
+        if (price) setSalesTaxAmount(applyRateToBigInt(price, tax.rate));
+      });
+    } else {
+      setSalesTaxRate(0);
+      setSalesTaxAmount("");
+    }
+  }, [isUsResident, usState, price]);
+
+  useEffect(() => {
+    if (selectedDomains) {
+      const calls = [
+        registrationCalls.approve(price),
+        ...registrationCalls.multiCallRenewal(
+          selectedDomainsToArray(selectedDomains),
+          duration
+        ),
+      ];
+
+      // If the user is a US resident, we add the sales tax
+      if (salesTaxRate) {
+        calls.unshift(registrationCalls.vatTransfer(salesTaxAmount)); // IMPORTANT: We use unshift to put the call at the beginning of the array
+      }
+
+      setCallData(calls);
+    }
+  }, [selectedDomains, price, salesTaxRate]);
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.card}>
+        <div className={styles.form}>
+          <div className="flex flex-col items-start gap-4 self-stretch">
+            <p className={styles.legend}>Your renewal</p>
+            <h3 className={styles.domain}>
+              {isStarkRootDomain(domain)
+                ? getDomainWithStark(domain)
+                : "Renew Your domain(s)"}
+            </h3>
+          </div>
+          <div className="flex flex-col items-start gap-6 self-stretch">
+            <TextField
+              helperText="Secure your domain's future and stay ahead with vital updates. Your email stays private with us, always."
+              label="Email address"
+              value={email}
+              onChange={(e) => changeEmail(e.target.value)}
+              color="secondary"
+              error={emailError}
+              errorMessage={"Please enter a valid email address"}
+              type="email"
+            />
+            <UsForm
+              isUsResident={isUsResident}
+              onUsResidentChange={() => setIsUsResident(!isUsResident)}
```

Comment:
it's up to you to decide but when I update a state value using its existing value I prefer to use the function form, so setIsUsResident(prev => !prev) for example.